test_physics_metrics.py

```diff
@@ -0,0 +1,1044 @@
+
+    # jets_pred2 = clusterJets(pt_pred2, eta, phi)
+    # print("pt_pred2", jets_pred2)
+    # performances_jet_pred2 = compareJets(jets_truth, jets_pred2)
+
+    # calculate MET and compare
+    met_truth = calculateMET(pt_truth, phi_truth)
+    met_puppi = calculateMET(pt_puppi, phi)
+    met_puppi_wcut = calculateMET(pt_puppi_wcut, phi)
+    # met_pred  = calculateMET(pt_pred,  phi)
+    # met_pred2 = calculateMET(pt_pred2, phi)
+    # print("***** one event ********")
+    # print("met truth", met_truth)
+    # print("met puppi", met_puppi)
+    # print("met puppi wcut", met_puppi_wcut)
+    # print("met pred", met_pred)
+
+    # evaluate the performances for the predictions
+    performances_jet_pred = []
+    mets_pred = []
+    neu_pred = []
+    neu_puppi = []
+    chlv_pred = []
+    chpu_pred = []
+    chlv_puppi = []
+    chpu_puppi = []
+
+    for pred in preds:
+        # print("preds: ", pred)
+        pred = np.array(pred[0][:, 0].cpu().detach())
+        #pred[pred<0.3] = 0
+        predcopy = pred
+        predcopyA = []
+        predcopyB = []
+        for j in range(len(predcopy)):
+            predcopyA.append(predcopy[j])
+
+        predcopy[charge_index] = -2
+        for m in range(len(predcopy)):
+            if predcopy[m]>-0.1:
+                neu_pred.append(predcopy[m])
+                neu_puppi.append(puppichg[m])
+        
+        predcopyA = np.array(predcopyA)
+        #predcopyA[predcopyA<0.3] = 0
+        #predcopyA[predcopyA>0.3] = 1
+
+
+        for mi in chglv_index:
+            chlv_pred.append(predcopyA[mi])
+            chlv_puppi.append(puppichg[mi])
+        for mj in chgpu_index:
+            chpu_pred.append(predcopyA[mj])
+            chpu_puppi.append(puppichg[mj])
+        # apply CHS to predictions
+        # charge_index = data.Charge_index[0]
+        #pred[charge_index] = puppichg[charge_index]
+        if testneu == 1:
+            predcopyA[charge_index] = puppi[charge_index]
+        
+        pt_pred = pt * predcopyA
+        jets_pred = clusterJets(pt_pred,  eta, phi, ifAK8)
+        njets_pred, pt_jets_pred, eta_jets_pred, phi_jets_pred, mass_jets_pred = ExtractJet(jets_pred)
+        performance_jet_pred = compareJets(jets_truth, jets_pred)
+
+        # MET
+        met_pred = calculateMET(pt_pred,  phi)
+
+        performances_jet_pred.append(performance_jet_pred)
+        mets_pred.append(met_pred)
+    
+    
+
+    return met_truth,performances_jet_CHS, performances_jet_puppi, met_puppi, performances_jet_puppi_wcut, met_puppi_wcut, performances_jet_pred, mets_pred, neu_pred, neu_puppi, chlv_pred, chpu_pred, chlv_puppi, chpu_puppi, njets_pf, njets_pred, njets_puppi, njets_truth, njets_CHS, pt_jets_pf, pt_jets_pred, pt_jets_puppi, pt_jets_truth, pt_jets_CHS, eta_jets_pf, eta_jets_pred, eta_jets_puppi, eta_jets_truth, eta_jets_CHS, phi_jets_pf, phi_jets_pred, phi_jets_puppi, phi_jets_truth, phi_jets_CHS, mass_jets_pf, mass_jets_pred, mass_jets_puppi, mass_jets_truth, mass_jets_CHS
```

Comment:
for longer term lets think about reducing the code duplicates by making a data class or something; for now it is fine.